apps/observable-sample/src/cat/cat.service.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Cat } from './entities/cat.entity';
+import { CatService } from './cat.service';
+import { of } from 'rxjs';
+
+const testCat1 = 'Test Cat 1';
+const testBreed1 = 'Test Breed 1';
+
+const catArray = [
+  new Cat(testCat1, testBreed1, 4),
+  new Cat('Test Cat 2', 'Test Breed 2', 3),
+  new Cat('Test Cat 3', 'Test Breed 3', 2),
+];
+
+const oneCat = new Cat(testCat1, testBreed1, 4);
+
+describe('CatService', () => {
+  let service: CatService;
+  let repo: Repository<Cat>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        CatService,
+        {
+          provide: getRepositoryToken(Cat),
+          useValue: {
+            find: jest.fn(() => of(catArray)),
+            findOne: jest.fn(() => of(oneCat)),
+            create: jest.fn(() => of(oneCat)),
+            save: jest.fn(() => of()),
+            update: jest.fn(() => of()),
+            delete: jest.fn(() => of()),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<CatService>(CatService);
+    repo = module.get<Repository<Cat>>(getRepositoryToken(Cat));
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+  describe('findAll()', () => {
+    it('should return an array of cats', () => {
+      service.findAll().subscribe((cats) => {
+        expect(cats).toEqual(catArray);
+      });
+    });
```

Comment:
Every test here should use a callback and in the `subscribe` there should be a call to the callback on complete
```suggestion
    it('should return an array of cats', (done) => {
      service.findAll().subscribe({
        next: (cats) => {
          expect(cats).toEqual(catArray);
        },
        complete: () => done()
      });
    });
```